contrib_bots/lib/converter.py

```diff
@@ -0,0 +1,228 @@
+                               'Converts `number` in the unit `unit_from` to ' +
+                               'the `unit_to` and prints the result\n' +
+                               '`number`: floating point number, e.g. 12, 13.05, 0.002\n' +
+                               '`unit_from` and `unit_to` are two of the following units:\n' +
+                               '* Bit, byte\n' +
+                               '* Kelvin (K), Fahrenheit (F), Celsius(C)\n' +
+                               '* Meter (m), Inch (in), Foot (ft), Yard (y),' +
+                               ' Mile (mi), Nautical-Mile (nmi)\n' +
+                               '* Square-Meter (m^2), Square-Kilometer (km^2),' +
+                               'Square-Centimeter (cm^2), Are, Hectare (ha),' +
+                               ' Square-Inch (in^2), Square-Foot (ft^2), Square-Yard (y^2), ' +
+                               'Acre, Square-Mile(mi^2)\n' +
+                               '* Cubic-Meter (m^3), Cubic-Centimeter (cm^3), Cubic-Decimeter, ' +
+                               'Cubic-Inch,' +
+                               ' Cubic-Foot (ft^3), Cubic-Yard (y^3), Fluid-Ounce, Liter (l) (\n' +
+                               '* Gram (g), Ton (t), Ounce (oz), Pound(lb)\n\n\n' +
+                               'Allowed prefixes are:\n' +
+                               '* atto, pico, femto, nano, micro, milli, centi, deci\n' +
+                               '* deca, hecto, kilo, mega, giga, tera, peta, exa\n\n\n' +
+                               'Usage examples:\n' +
+                               '* `@convert 12 celsius fahrenheit`\n' +
+                               '* `@convert 0.002 kilomile millimeter`\n' +
+                               '* `@convert 31000 attocubic-decimeter megaounce`\n' +
+                               '* `@convert 56 g lb`\n'
+                               )
+                continue
+            if (idx+3) < len(words):
+                number = words[idx+1]
+                unit_from = self.units.aliases.get(words[idx+2], words[idx+2])
+                unit_to = self.units.aliases.get(words[idx+3], words[idx+3])
+                exponent = 0
+
+                if not self.units.isfloat(number):
+                    results.append(number + ' is not a valid number. ' +
+                                   ' Enter `@convert help` for help on using the converter.')
+                    continue
+
+                number = float(number)
+                number_res = copy.copy(number)
+
+                for key, exp in self.units.prefixes.items():
+                    if unit_from.startswith(key):
+                        exponent += exp
+                        unit_from = unit_from[len(key):]
+                    if unit_to.startswith(key):
+                        exponent -= exp
+                        unit_to = unit_to[len(key):]
+
+                uf_to_std = self.units.units.get(unit_from, False)
```

Comment:
Hm, I use the get to prevent Exceptions and set a default value for ut_to_std and uf_to_std, so not sure if I should change this; it would make the code longer eventually.